eng/pipelines/runtime.yml

```diff
@@ -609,7 +609,6 @@ extends:
           - osx_x64
           - linux_arm64
           - windows_arm64
-          - osx_arm64
```

Comment:
osx_arm64 is still covered by `# CoreCLR NativeAOT release build and libraries tests`, so this removal is not significant coverage loss.